yt_dlp/extractor/ard.py

```diff
@@ -357,14 +358,25 @@ def _extract_episode_info(self, title):
         }), get_all=False)
 
     def _real_extract(self, url):
-        video_id, display_id = self._match_valid_url(url).group('id', 'display_id')
+        display_id = self._match_id(url)
 
         page_data = self._download_json(
-            f'https://api.ardmediathek.de/page-gateway/pages/ard/item/{video_id}', video_id, query={
+            f'https://api.ardmediathek.de/page-gateway/pages/ard/item/{display_id}', display_id, query={
                 'embedded': 'false',
                 'mcV6': 'true',
             })
 
+        # For user convenience we use the old contentId instead of the longer crid
+        # Ref: https://github.com/yt-dlp/yt-dlp/issues/8731#issuecomment-1874398283
+        old_id = traverse_obj(page_data, ('tracking', 'atiCustomVars', 'contentId', {str_or_none}))
+        if old_id:
+            video_id = old_id
+            archive_ids = [make_archive_id(ARDBetaMediathekIE, video_id)]
```

Comment:
We have to use the new id here
```suggestion
            archive_ids = [make_archive_id(ARDBetaMediathekIE, diplay_id)]
```